test-clients/src/main/java/com/hedera/services/bdd/suites/contract/BigArraySpec.java

```diff
@@ -0,0 +1,108 @@
+ * limitations under the License.
+ * ‍
+ */
+
+import com.hedera.services.bdd.spec.HapiApiSpec;
+import com.hedera.services.bdd.spec.transactions.TxnVerbs;
+import com.hedera.services.bdd.spec.utilops.CustomSpecAssert;
+import com.hedera.services.bdd.suites.HapiApiSuite;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ThreadLocalRandom;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.getContractInfo;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.contractCall;
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.contractCallLocal;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.fileUpdate;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.withOpContext;
+
+public class BigArraySpec extends HapiApiSuite {
+	private static final Logger log = LogManager.getLogger(BigArraySpec.class);
+
+	private static final String BA_CHANGEARRAY_ABI = "{\"constant\":false,\"inputs\":[{\"name\":\"_value\",\"type\":\"uint256\"}],\"name\":\"changeArray\",\"outputs\":[],\"payable\":false,\"stateMutability\":\"nonpayable\",\"type\":\"function\"}";
+	private static final String BA_GROWTO_ABI = "{\"constant\":false,\"inputs\":[{\"internalType\":\"uint256\",\"name\":\"_limit\",\"type\":\"uint256\"}],\"name\":\"growTo\",\"outputs\":[],\"payable\":false,\"stateMutability\":\"nonpayable\",\"type\":\"function\"}";
+	private static final String PATH_TO_BIG_ARRAY_BYTECODE = "src/main/resource/testfiles/GrowArray.bin";
+
+	public static void main(String... args) {
+		/* Has a static initializer whose behavior seems influenced by initialization of ForkJoinPool#commonPool. */
+		new org.ethereum.crypto.HashUtil();
+
+		new BigArraySpec().runSuiteAsync();
+	}
+
+	@Override
+	protected Logger getResultsLogger() {
+		return log;
+	}
+
+	@Override
+	protected List<HapiApiSpec> getSpecsInSuite() {
+		return List.of(new HapiApiSpec[] {
+				bigArray(),
+		});
+	}
+
+	HapiApiSpec bigArray() {
+		var MAX_CONTRACT_STORAGE_ALLOWED = 1025;
+
+		return HapiApiSpec.defaultHapiSpec("BigArray")
+				.given(
+						fileUpdate(APP_PROPERTIES)
+								.overridingProps(Map.of(
+										"contracts.maxStorageKb", "" + MAX_CONTRACT_STORAGE_ALLOWED
+								))
+								.payingWith(GENESIS),
+						TxnVerbs.fileCreate("bigArrayContractFile")
+								.path(PATH_TO_BIG_ARRAY_BYTECODE),
+						TxnVerbs.contractCreate("bigArrayContract")
+								.bytecode("bigArrayContractFile")
+				)
+				.when(
+						withOpContext((spec, opLog) -> {
+							int kbPerStep = 16;
+
+							for (int sizeNow = kbPerStep; sizeNow < MAX_CONTRACT_STORAGE_ALLOWED; sizeNow += kbPerStep) {
+								var subOp1 = contractCall(
+										"bigArrayContract", BA_GROWTO_ABI, sizeNow)
+										.gas(300_000L)
+										.logged();
+								CustomSpecAssert.allRunFor(spec, subOp1);
```

Comment:
It's probably more efficient to change the code snippet to 
```
							List<HapiSpecOperation> subOps = new ArrayList<>();
							for (int sizeNow = 0; sizeNow <= MAX_CONTRACT_STORAGE_ALLOWED; sizeNow += kbPerStep) {
								var subOp1 = contractCall(
										"bigArrayContract", BA_GROWTO_ABI, sizeNow);
								subOps.add(subOp1);
							}
							CustomSpecAssert.allRunFor(spec, subOps);
```

Reply:
Done